src/components/snippetpanel.ts

```diff
@@ -0,0 +1,230 @@
+
+    private async loadSnippets() {
+        const snipetsFile = path.join(this.extension.extensionRoot, 'resources', 'snippetpanel', 'snippetpanel.json')
+        const snippets: {
+            mathSymbols: {
+                [category: string]: {
+                    name: string;
+                    keywords?: string;
+                    category?: string;
+                    source: string;
+                    snippet: string;
+                    svg?: string;
+                    shrink?: boolean;
+                }[];
+            };
+        } = JSON.parse(readFileSync(snipetsFile, { encoding: 'utf8' }))
+
+        const mathSymbolPromises: Promise<any>[] = []
+        for (const category in snippets.mathSymbols) {
+            for (let i = 0; i < snippets.mathSymbols[category].length; i++) {
+                const symbol = snippets.mathSymbols[category][i]
+                if (symbol.svg === undefined) {
+                    mathSymbolPromises.push(
+                        new Promise((resolve, reject) => {
+                            this.mathJax
+                                .typeset({
+                                    math: symbol.source,
+                                    format: 'TeX',
+                                    svgNode: true
+                                })
+                                .then(
+                                    (data: {
+                                        height: string;
+                                        speakText: string;
+                                        style: string;
+                                        svgNode: SVGSVGElement;
+                                        width: string;
+                                    }) => {
+                                        let svg = data.svgNode.outerHTML
+                                        svg = svg.replace(
+                                            /<title([^>]*)>(.*)<\/title>/,
+                                            `<title$1>${symbol.name.toLocaleUpperCase()}.${
+                                                symbol.keywords ? ' Keywords: ' + symbol.keywords : ''
+                                            }</title>`
+                                        )
+                                        if (symbol.shrink) {
+                                            svg = svg.replace(/^<svg/, '<svg class="shrink"')
+                                        }
+                                        symbol.svg = svg
+                                        resolve()
+                                    }
+                                )
+                                .catch(reject)
+                        })
+                    )
+                }
+            }
+        }
+        Promise.all(mathSymbolPromises).finally(() => {
+            if (mathSymbolPromises.length > 0) {
+                writeFileSync(snipetsFile, JSON.stringify(snippets, undefined, 4))
```

Comment:
For the release I'll just commit the version with the svg parameter saved, however since this is needed for development, and shouldn't be executed by users as long as they get the with-svgs version I don't think anything needs to be done with that. Does this sounds good to you?

Reply:
Please extract the code generating the JSON file from `snippetpanel.ts` and save it into a file in `LaTeX-Workshop/dev`. We should not include the code in the release.

Reply:
Done, see ad93623fd2c4f1ad9bab673d50cf824504257a71